src/test/compile-fail/macro-context.rs

```diff
@@ -29,5 +28,5 @@ fn main() {
         m!() => {}  //~ NOTE the usage of `m!` is likely invalid in pattern context
     }
 
-    m!();           //~ NOTE the usage of `m!` is likely invalid in statement context
+    m!();
```

Comment:
In this test we get fewer errors because we panic a little bit earlier. Note that if we remove the other error causing lines, we still get an error for line 31, so it's not a regression.